spock-core/src/main/java/org/spockframework/compat/groovy2/GroovyCodeVisitorCompat.java

```diff
@@ -16,22 +16,19 @@
  *  specific language governing permissions and limitations
  *  under the License.
  */
-package groovy.transform;
+package org.spockframework.compat.groovy2;
 
-import java.lang.annotation.ElementType;
-import java.lang.annotation.Retention;
-import java.lang.annotation.RetentionPolicy;
-import java.lang.annotation.Target;
+import org.codehaus.groovy.ast.expr.LambdaExpression;
+import org.codehaus.groovy.ast.expr.MethodReferenceExpression;
 
 /**
- * WARNING: This is only present here so that we can build with 2.4 as well, the code will not be shipped.
- *
- * Collector annotation for {@link NamedParam}.
- *
- * @since 2.5.0
+ * Compatibility layer to keep codebase compilable with both Groovy 2.5 and 3.0
  */
-@Retention(RetentionPolicy.RUNTIME)
-@Target(ElementType.PARAMETER)
-public @interface NamedParams {
-  NamedParam[] value();
+public interface GroovyCodeVisitorCompat {
+
+  default void visitMethodReferenceExpression(MethodReferenceExpression expr) {
```

Comment:
why are they default methods, the only place where we use them, we also implement those methods

Reply:
yup, they can be changed to ordinary methods.